morph_tool/spatial.py

```diff
@@ -27,3 +34,47 @@ def point_to_section_segment(neuron, point):
             return (section.id + 1, offset[0][0])
 
     raise ValueError('Cannot find point in morphology that matches: {}'.format(point))
+
+
+def align_pyr_morph(morph, direction=None):
+    """In place alignment of a pyramidal morphology such that apical tree is along 'direction'.
+
+    The algorithm is based on SVD decomposititon of the correlation matrix obtained from all points
+    in apical neurites, giving the principal axis of apical dendrite.
+    If not apicals are present, no rotation is applied.
+
+    Args:
+        morph (morphio.Morphology): morphology to align
+        direction (ndarray): 3 vector with for final direction, if None, [0, 1, 0] will be used
+    """
+    if direction is None:
+        direction = [0.0, 1.0, 0.0]
+    apical_points = []
+    for section in morph.iter():
+        if section.type == SectionType.apical_dendrite:
+            apical_points += list(section.points)
+
+    if len(apical_points) == 0:
+        return None
```

Comment:
Maybe a more specific value should be returned here, like `False` for example. Because here we can not check what is returned to know if it worked or not. Or maybe we can return `None` here and return the rotation angle or the initial direction that was detected by the SVD. So the user can check what the function did.

Reply:
yes, I return rotation matrix, this is in fact important to have!